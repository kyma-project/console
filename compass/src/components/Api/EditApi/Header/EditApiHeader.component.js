import React from 'react';
import PropTypes from 'prop-types';
import LuigiClient from '@kyma-project/luigi-client';

import { ActionBar } from 'fundamental-react';
import {
  Button,
  Breadcrumb,
  Panel,
  PanelGrid,
} from '@kyma-project/react-components';

EditApiHeader.propTypes = {
  apiData: PropTypes.object.isRequired,
  applicationName: PropTypes.string.isRequired,
  saveChanges: PropTypes.func.isRequired,
  canSaveChanges: PropTypes.bool.isRequired,

  deleteApi: PropTypes.func.isRequired,
  deleteEventApi: PropTypes.func.isRequired,
  sendNotification: PropTypes.func.isRequired,
};

export default function EditApiHeader({
  apiData,
  applicationName,
  saveChanges,
  canSaveChanges,

  deleteApi,
  deleteEventApi,
  sendNotification,
}) {
  function navigateBackToApplication() {
    LuigiClient.linkManager()
      .fromContext('application')
      .navigate('');
  }

  function navigateToApplications() {
    LuigiClient.linkManager()
      .fromContext('tenant')
      .navigate('/applications');
  }

  function handleDelete() {
    LuigiClient.uxManager()
      .showConfirmationModal({
        header: `Delete ${apiData.generalInformation.name}`,
        body: `Are you sure you want to delete "${apiData.generalInformation.name}"?`,
        buttonConfirm: 'Delete',
        buttonDismiss: 'Cancel',
      })
      .then(() => {
        deleteEntry();
      })
      .catch(() => {});
  }

  async function deleteEntry() {
    const id = apiData.id;
    const name = apiData.generalInformation.name;

    try {
      if (apiData.apiType === 'API') {
        await deleteApi(id);
      } else {
        await deleteEventApi(id);
      }
      sendNotification({
        variables: {
          content: `"${name}" deleted from application.`,
          title: `${name}`,
          color: '#359c46',
          icon: 'accept',
          instanceName: name,
        },
      });
      LuigiClient.uxManager().setDirtyStatus(false);
      navigateBackToApplication();
    } catch (error) {
      console.warn(error);
      LuigiClient.uxManager().showAlert({
        text: error.message,
        type: 'error',
        closeAfter: 10000,
      });
    }
  }

  function formatSpecType() {
    switch (apiData.spec.type) {
      case 'ASYNC_API':
        return 'Event API';
      case 'OPEN_API':
        return 'OpenAPI';
      case 'ODATA':
        return 'OData';
      default:
        return '';
    }
  }

  return (
    <header className="fd-has-background-color-background-2">
      <section className="fd-has-padding-regular fd-has-padding-bottom-none action-bar-wrapper">
        <section>
          <Breadcrumb>
            <Breadcrumb.Item
              name="Applications"
              url="#"
              onClick={navigateToApplications}
            />
            <Breadcrumb.Item
              name={applicationName}
              url="#"
              onClick={navigateBackToApplication}
            />
            <Breadcrumb.Item />
          </Breadcrumb>
          <ActionBar.Header title={`Edit ${apiData.generalInformation.name}`} />
        </section>
        <ActionBar.Actions>
          <Button
            onClick={saveChanges}
            disabled={!canSaveChanges}
            option="emphasized"
          >
            Save
          </Button>
          <Button onClick={handleDelete} option="light" type="negative">
            Delete
          </Button>
        </ActionBar.Actions>
      </section>
      <PanelGrid nogap cols={4}>
        <Panel>
          <Panel.Body>
            <p className="fd-has-color-text-4 fd-has-margin-bottom-none">
              Type
            </p>
            {formatSpecType()}
          </Panel.Body>
        </Panel>
      </PanelGrid>
    </header>
  );
}
