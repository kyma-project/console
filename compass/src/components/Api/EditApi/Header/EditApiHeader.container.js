import { graphql } from 'react-apollo';
import { compose } from 'recompose';

import { SEND_NOTIFICATION } from './../../../../gql';
import { DELETE_EVENT_API, DELETE_API } from '../../../Application/gql';

import EditApiHeader from './EditApiHeader.component';

export default compose(
  graphql(SEND_NOTIFICATION, {
    name: 'sendNotification',
  }),
  graphql(DELETE_API, {
    props: props => ({
      deleteApi: async apiId => {
        await props.mutate({ variables: { id: apiId } });
      },
    }),
  }),
  graphql(DELETE_EVENT_API, {
    props: props => ({
      deleteEventApi: async eventApiId => {
        await props.mutate({ variables: { id: eventApiId } });
      },
    }),
  }),
)(EditApiHeader);
