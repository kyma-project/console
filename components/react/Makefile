ci-pr: resolve build validate test
ci-master: resolve build validate test
ci-release: resolve build validate test

resolve:
	npm install

build:
	npm run build

validate:
	npm run lint-check
	eslint -c .eslinterrc ./src

test:
	CI=true npm run test
	