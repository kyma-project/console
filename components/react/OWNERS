reviewers:
    - default-reviewers

approvers:
    - default-approvers
    - Wawrzyn321
    - content-reviewers