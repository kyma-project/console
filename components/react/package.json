{
  "name": "@kyma-project/react-components",
  "version": "0.0.43",
  "main": "dist/index.js",
  "private": false,
  "license": "Apache-2.0",
  "directories": {
    "dist": "src"
  },
  "scripts": {
    "start:kyma": "npm run build:watch",
    "start:kyma:api": "npm run build:watch",
    "build": "webpack --config webpack.config.prod.js",
    "build:watch": "webpack -w --config webpack.config.dev.js",
    "test": "jest src/**; exit 0",
    "test:watch": "jest src/** --watch",
    "test:watchAll": "jest src/** --watchAll",
    "test:coverage:report": "open ./coverage/lcov-report/index.html",
    "lint-fix": "prettier --config ../../.prettierrc --write \"./src/**/*.{js,jsx,ts,tsx,html,css,json}\"",
    "lint-check": "prettier --config ../../.prettierrc --check \"./src/**/*.{js,jsx,ts,tsx,html,css,json}\""
  },
  "dependencies": {
    "@babel/runtime-corejs2": "^7.12.5",
    "lodash": "^4.17.21",
    "polished": "^1.8.1",
    "styled-components-modifiers": "^1.0.1"
  },
  "devDependencies": {
    "@babel/types": "7.0.0-beta.44",
    "ajv": "^6.12.3",
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.0",
    "babel-eslint": "^8.0.1",
    "babel-jest": "^22.4.1",
    "babel-loader": "^7.1.2",
    "babel-polyfill": "^6.26.0",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "enzyme": "^3.1.0",
    "enzyme-adapter-react-16": "^1.0.2",
    "exec-sh": "^0.3.3",
    "file-loader": "^1.1.11",
    "fiori-fundamentals": "1.5.3",
    "fontfaceobserver": "^2.0.13",
    "handlebars": "^4.7.3",
    "jest": "^24.0.0",
    "jest-styled-components": "^4.11.0-1",
    "npm-run-all": "^4.1.1",
    "raf": "^3.4.0",
    "react-jsonschema-form": "1.7.0",
    "react-modal": "^3.5.1",
    "react-scrollbar": "^0.5.4",
    "react-test-renderer": "^16.0.0",
    "style-loader": "^0.20.2",
    "styled-components": "^3.2.3",
    "styled-components-grid": "^2.2.0",
    "webpack": "^4.28.1",
    "webpack-cli": "^3.2.1",
    "webpack-merge": "^4.2.1"
  },
  "peerDependencies": {
    "styled-components": "^3.2.3"
  },
  "resolutions": {
    "uglify-es": "3.3.7"
  },
  "jest": {
    "collectCoverage": true,
    "collectCoverageFrom": [
      "src/**",
      "!src/**/*.snap",
      "!src/components/index.js",
      "!src/modifiers/**",
      "!src/index.js"
    ],
    "moduleNameMapper": {
      "^.+\\.(css|less|scss)$": "babel-jest"
    },
    "moduleDirectories": [
      "node_modules",
      "./"
    ],
    "setupFiles": [
      "raf/polyfill",
      "./test_setup/config"
    ]
  }
}
