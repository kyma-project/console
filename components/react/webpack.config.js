const webpack = require('webpack');
const path = require('path');
const pkg = require('./package.json');

require('babel-polyfill');

let libraryName = pkg.name;

module.exports = {
  entry: ['./src/index'],
  module: {
    loaders: [
      { test: /\.js?$/, loader: 'babel-loader', exclude: /node_modules/ },
      {
        test: /\.css$/,
        use: [{ loader: 'style-loader' }, { loader: 'css-loader' }],
      },
      {
        test: /\.(eot|svg|ttf|woff|woff2|otf)$/,
        loader: 'file-loader?name=fonts/[name].[ext]',
      },
    ],
  },
  resolve: {
    modules: [path.resolve('./node_modules'), 'node_modules'],
    extensions: ['.js'],
  },
  output: {
    path: path.join(__dirname, '/lib'),
    publicPath: '/',
    filename: 'index.js',
    library: libraryName,
    libraryTarget: 'umd',
    umdNamedDefine: true,
  },
  devtool: 'source-map',
  devServer: {
    contentBase: './src',
    hot: true,
  },
  plugins: [
    new webpack.optimize.OccurrenceOrderPlugin(),
    new webpack.HotModuleReplacementPlugin(),
    new webpack.NoEmitOnErrorsPlugin(),
  ],
  externals: {
    'styled-components': {
      commonjs: 'styled-components',
      commonjs2: 'styled-components',
      amd: 'styled-components',
    },
  },
};
