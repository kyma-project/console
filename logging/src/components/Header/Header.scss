.header__settings-group {
  display: flex;
  padding-bottom: 8px;
  border-bottom: 1px solid var(--fd-color-background-1);

  & > * {
    margin-right: 16px;
  }

  @media (max-width: 500px) {
    display: initial;
    & > * {
      flex: 1;
    }
  }
}

.header__settings-group__toggle {
  padding-top: 30px;
}

.header-toolbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
