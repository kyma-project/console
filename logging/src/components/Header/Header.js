import React, { useContext } from 'react';
import './Header.scss';

import { Panel } from 'fundamental-react';

import SearchInput from './../Shared/SearchInput/SearchInput';
import LabelsInput from './../Shared/LabelsInput/LabelsInput';
import AdvancedSettings from './../Shared/AdvancedSettings/AdvancedSettings';
import LabelsDisplay from './../Shared/LabelsDisplay/LabelsDisplay';
import OptionsDropdown from './../Shared/SelectDropdown/SelectDropdown';
import AutoRefreshButton from './../Shared/AutoRefreshButton/AutoRefreshButton';
import { PERIODS, SORT_TYPES } from '../../constants';
import { SearchParamsContext } from '../Logs/SearchParams.reducer';

export default function Header() {
  const [advancedShown, setAdvancedShown] = React.useState(false);
  const [state, actions] = useContext(SearchParamsContext);

  const { logsPeriod, sortDirection } = state;

  function toggleAdvancedSettingsVisibility() {
    setAdvancedShown(!advancedShown);
  }

  const advancedSettingsButtonText = advancedShown
    ? 'Hide Advanced Settings'
    : 'Show Advanced Settings';

  return (
    <Panel className="fd-has-padding-regular fd-has-padding-bottom-none">
      <h1 className="fd-has-type-3 fd-has-padding-bottom-tiny">Logs</h1>
      <section className="header__settings-group">
        <LabelsInput />
        <SearchInput />
        <span
          data-test-id="advanced-settings-toggle"
          className="link-button fd-has-type-minus-1 header__settings-group__toggle"
          onClick={toggleAdvancedSettingsVisibility}
          role="button"
        >
          {advancedSettingsButtonText}
        </span>
      </section>
      {advancedShown && (
        <AdvancedSettings hideSettings={() => setAdvancedShown(false)} />
      )}

      <div>
        <LabelsDisplay />
        <div className="header__options-wrapper">
          <AutoRefreshButton />
          <OptionsDropdown
            availabelValues={PERIODS}
            icon="past"
            currentValue={logsPeriod}
            updateValue={actions.setLogsPeriod}
          />
          <OptionsDropdown
            availabelValues={SORT_TYPES}
            icon="sort"
            currentValue={sortDirection}
            updateValue={actions.setSortDir}
          />
        </div>
      </div>
    </Panel>
  );
}
