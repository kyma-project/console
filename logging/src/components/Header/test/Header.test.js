import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Header from './../Header';
import {
  actions,
  defaultSearchParams,
  SearchParamsContext,
} from '../../Logs/SearchParams.reducer';

jest.mock('@kyma-project/common', () => ({
  getApiUrl: () => 'kyma.local',
}));

describe('Header', () => {
  it('Shows / hides advanced settings when corresponding button is clicked', () => {
    const mockActions = {
      setLogsPeriod: jest.fn(),
      setSortDir: jest.fn(),
    };

    const { queryByText } = render(
      <SearchParamsContext.Provider value={[defaultSearchParams, mockActions]}>
        <Header />
      </SearchParamsContext.Provider>,
    );

    // should be initially hidden
    const advancedOptionsToggle = queryByText('Show Advanced Settings');
    expect(advancedOptionsToggle).toBeInTheDocument();
    expect(queryByText('Advanced Settings')).not.toBeInTheDocument();

    // toggle
    fireEvent.click(advancedOptionsToggle);
    expect(advancedOptionsToggle).toHaveTextContent('Hide Advanced Settings');
    expect(queryByText('Advanced Settings')).toBeInTheDocument();
  });
});
