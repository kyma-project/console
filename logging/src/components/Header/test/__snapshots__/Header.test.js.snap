// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders with minimal props 1`] = `
<div
  className="fd-panel fd-has-padding-bottom-none header"
>
  <h1
    className="fd-has-type-3 fd-has-padding-bottom-tiny fd-has-padding-top-regular fd-has-padding-right-regular fd-has-padding-left-regular "
  >
    Logs
  </h1>
  <section
    className="header__settings-group fd-has-padding-right-regular fd-has-padding-left-regular "
  >
    <section
      className="fd-display-l-inline-block"
    >
      <span
        className="caption-muted"
      >
        Labels
      </span>
      <div
        className="fd-combobox-input"
      >
        <div
          className="fd-popover"
        >
          <div
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="fd-popover__control"
            >
              <div
                className="fd-combobox-control"
                onClick={[Function]}
              >
                <div
                  className="fd-input-group fd-input-group--after"
                >
                  <input
                    className="fd-input"
                    placeholder="Select Label"
                    type="text"
                  />
                  <span
                    className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button"
                  >
                    <button
                      className=" fd-button--light sap-icon--navigation-down-arrow"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      className="fd-has-margin-right-small"
    >
      <span
        className="caption-muted search-input__caption"
      >
        Search
        <div
          className="small-inline-help-wrapper"
        >
          <span
            className="fd-inline-help"
          >
            <span
              className="fd-inline-help__content fd-inline-help__content--right"
            >
              Search for logs by text (optional)
            </span>
          </span>
        </div>
      </span>
      <input
        className=""
        id="search-input"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value="search phrase"
      />
    </section>
    <span
      className="link-button fd-has-type-minus-1 header__settings-group__toggle"
      data-test-id="advanced-settings-toggle"
      onClick={[Function]}
      role="button"
    >
      Show Advanced Settings
    </span>
  </section>
  <div
    className="header-toolbar fd-has-padding-right-regular fd-has-padding-left-regular "
  >
    <section
      className="labels-display"
    >
      <ul>
        <li>
          <span
            className="fd-token caption-muted y-fd-token--no-button"
            role="button"
          >
            a
          </span>
        </li>
        <li>
          <span
            className="fd-token caption-muted y-fd-token--no-button"
            role="button"
          >
            b
          </span>
        </li>
        <li>
          <span
            className="fd-token caption-muted"
            onClick={[Function]}
            role="button"
          >
            a
          </span>
        </li>
        <li>
          <span
            className="fd-token caption-muted"
            onClick={[Function]}
            role="button"
          >
            b
          </span>
        </li>
      </ul>
      <span
        className="link-button fd-has-type-minus-1 fd-has-margin-left-tiny"
        data-test-id="clear-all"
        onClick={[Function]}
      >
        Clear All
      </span>
    </section>
    <div>
      <button
        className="fd-button--light sap-icon--media-pause link-button fd-has-margin-right-tiny fd-has-type-minus-1"
        data-test-id="auto-refresh-button"
        onClick={[Function]}
        size="xs"
      >
        auto refresh
      </button>
      <span
        className="link-button fd-has-type-minus-1 select-dropdown"
      >
        <div
          className="fd-popover"
        >
          <div
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="fd-popover__control"
            >
              <button
                className="fd-button--light sap-icon--past fd-has-margin-right-tiny"
                onClick={[Function]}
                size="xs"
              >
                last 15 minutes
              </button>
            </div>
          </div>
        </div>
      </span>
      <span
        className="link-button fd-has-type-minus-1 select-dropdown"
      >
        <div
          className="fd-popover"
        >
          <div
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="fd-popover__control"
            >
              <button
                className="fd-button--light sap-icon--sort fd-has-margin-right-tiny"
                onClick={[Function]}
                size="xs"
              >
                newest first
              </button>
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
</div>
`;

exports[`Header Shows / hides advanced settings when corresponding button is clicked 1`] = `
<Panel
  className="fd-has-padding-bottom-none header"
>
  <h1
    className="fd-has-type-3 fd-has-padding-bottom-tiny fd-has-padding-top-regular fd-has-padding-right-regular fd-has-padding-left-regular "
  >
    Logs
  </h1>
  <section
    className="header__settings-group fd-has-padding-right-regular fd-has-padding-left-regular "
  >
    <LabelsInput
      maxRecentCount={4}
    />
    <SearchInput
      compact={false}
    />
    <span
      className="link-button fd-has-type-minus-1 header__settings-group__toggle"
      data-test-id="advanced-settings-toggle"
      onClick={[Function]}
      role="button"
    >
      Hide Advanced Settings
    </span>
  </section>
  <AdvancedSettings
    hideSettings={[Function]}
  />
  <div
    className="header-toolbar fd-has-padding-right-regular fd-has-padding-left-regular "
  >
    <LabelsDisplay />
    <div>
      <AutoRefreshButton />
      <SelectDropdown
        availabelValues={
          Array [
            "last minute",
            "last 5 minutes",
            "last 15 minutes",
            "last hour",
          ]
        }
        compact={false}
        currentValue="last 15 minutes"
        icon="past"
        updateValue={[Function]}
      />
      <SelectDropdown
        availabelValues={
          Array [
            Object {
              "label": "newest first",
              "value": "descending",
            },
            Object {
              "label": "oldest first",
              "value": "ascending",
            },
          ]
        }
        compact={false}
        currentValue="descending"
        icon="sort"
        updateValue={[Function]}
      />
    </div>
  </div>
</Panel>
`;

exports[`Header Shows / hides advanced settings when corresponding button is clicked 2`] = `
<Panel
  className="fd-has-padding-bottom-none header"
>
  <h1
    className="fd-has-type-3 fd-has-padding-bottom-tiny fd-has-padding-top-regular fd-has-padding-right-regular fd-has-padding-left-regular "
  >
    Logs
  </h1>
  <section
    className="header__settings-group fd-has-padding-right-regular fd-has-padding-left-regular "
  >
    <LabelsInput
      maxRecentCount={4}
    />
    <SearchInput
      compact={false}
    />
    <span
      className="link-button fd-has-type-minus-1 header__settings-group__toggle"
      data-test-id="advanced-settings-toggle"
      onClick={[Function]}
      role="button"
    >
      Hide Advanced Settings
    </span>
  </section>
  <AdvancedSettings
    hideSettings={[Function]}
  />
  <div
    className="header-toolbar fd-has-padding-right-regular fd-has-padding-left-regular "
  >
    <LabelsDisplay />
    <div>
      <AutoRefreshButton />
      <SelectDropdown
        availabelValues={
          Array [
            "last minute",
            "last 5 minutes",
            "last 15 minutes",
            "last hour",
          ]
        }
        compact={false}
        currentValue="last 15 minutes"
        icon="past"
        updateValue={[Function]}
      />
      <SelectDropdown
        availabelValues={
          Array [
            Object {
              "label": "newest first",
              "value": "descending",
            },
            Object {
              "label": "oldest first",
              "value": "ascending",
            },
          ]
        }
        compact={false}
        currentValue="descending"
        icon="sort"
        updateValue={[Function]}
      />
    </div>
  </div>
</Panel>
`;
